src/styles/main.scss

```diff
@@ -1,3 +1,58 @@
 body {
   margin: 0;
 }
+
+$day-size: 100px;
+$border-width: 1px;
+$font-size: 30px;
+$font-family: arial, sans-serif;
+$padding: 10px;
+$columns: 7;
+$calendar-width: ($columns * $day-size)
+  + ($border-width * ($columns - 1)) + ($padding * 2);
+
+.days {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $border-width;
+  margin: 40px;
+  width: $calendar-width;
+
+  &__day {
+    display: flex;
+    width: $day-size;
+    height: $day-size;
+    border: $border-width solid black;
+    background-color: #eee;
+    align-items: center;
+    justify-content: center;
+    transition: transform 500ms;
+
+    &:hover {
+      background-color: #ff4fa7;
+      transform: translateY(-20px);
+      cursor: pointer;
+    }
+  }
+
+  @for $dayOfMonth from 1 through 31 {
+    .days__day:nth-child(#{$dayOfMonth})::before {
+      content: "#{$dayOfMonth}";
+      font-family: Arial, sans-serif;
+      font-size: $font-size;
```

Comment:
It is not necessary to set the styles for the text in a loop, it can be done once for the required class.